rd-net/Lifetimes/Serialization/UnsafeWriter.cs

```diff
@@ -24,13 +24,20 @@ public sealed unsafe class UnsafeWriter
     public struct Cookie : IDisposable
     {      
       private readonly UnsafeWriter myWriter;
+      private readonly Action myCleanupAction;
 
       private readonly int myStart;
 
       [MethodImpl(MethodImplAdvancedOptions.AggressiveInlining)]
-      public Cookie(UnsafeWriter writer) : this()
+      public Cookie(UnsafeWriter writer) : this(writer, null)
+      {
+      }
+
+      [MethodImpl(MethodImplAdvancedOptions.AggressiveInlining)]
+      public Cookie(UnsafeWriter writer, Action cleanupAction = null) : this()
```

Comment:
It's kind of dangerous to give user ability to invoke random cleanup actions after cookie is dispose. Do you really need such a power in public API?

Reply:
I do not see any reason for the Cookie to have public ctor in the first place. I do understand that it was created public t oavoid memory allocations

Reply:
Yes, I think memory allocaton is the main reason.